pkg/template/core/go_value.go

```diff
@@ -28,6 +28,9 @@ func (e GoValue) AsValueWithCheckedMapKeys(forceMapStringKeys bool) interface{}
 }
 
 func (e GoValue) asStarlarkValue(val interface{}) starlark.Value {
```

Comment:
why do we need this change in this PR?

Reply:
This is not strictly needed for this PR, but it makes sense to have this conversion possible. It will be needed to convert data value objects in #32.

Reply:
i think it should be part of that PR. i would like to make this PR as compact as possible and kick future work into future PRs and discussions.

Reply:
This is removed from the PR now.